adb/adb_commands.py

```diff
@@ -30,218 +30,307 @@
+        """
+        connection = self.protocol_handler.Open(self.handle, destination=b'sync:')
+        listing = self.filesync_handler.List(connection, device_path)
+        connection.Close()
+        return listing
+
+    def Reboot(self, destination=b''):
+        """Reboot the device.
+
+        Args:
+          destination: Specify 'bootloader' for fastboot.
+        """
+
+        self.protocol_handler.Open(self.handle, b'reboot:%s' % destination)
+
+    def RebootBootloader(self):
+        """Reboot device into fastboot."""
+        self.Reboot(b'bootloader')
+
+    def Remount(self):
+        """Remount / as read-write."""
+
+        return self.protocol_handler.Command(self.handle, service=b'remount')
+
+    def Root(self):
+        """Restart adbd as root on the device."""
+
+        return self.protocol_handler.Command(self.handle, service=b'remount')
+
+    def Shell(self, command, timeout_ms=None):
+        """Run command on the device, returning the output.
+
+        Args:
+          command: Shell command to run
+          timeout_ms: Maximum time to allow the command to run.
+        """
+
+        return self.protocol_handler.Command(self.handle, service=b'shell', command=command, timeout_ms=timeout_ms)
+
+    def StreamingShell(self, command, timeout_ms=None):
+        """Run command on the device, yielding each line of output.
+
+        Args:
+          command: Command to run on the target.
+          timeout_ms: Maximum time to allow the command to run.
+
+        Yields:
+          The responses from the shell command.
+        """
+        return self.protocol_handler.StreamingCommand(
+            self.handle, service=b'shell', command=command,
+            timeout_ms=timeout_ms)
+
+    def Logcat(self, options, timeout_ms=None):
+        """Run 'shell logcat' and stream the output to stdout.
+
+        Args:
+          options: Arguments to pass to 'logcat'.
+          timeout_ms: Maximum time to allow the command to run.
+        """
+        return self.StreamingShell('logcat %s' % options, timeout_ms)
+
+    def InteractiveShell(self, command=None, strip_command=True, delimiter=None, strip_delimiter=True):
+        """Get stdout from the currently open interactive shell and optionally run a command
+        on the device, returning all output.
+
+        Args:
+          command: Optional. Command to run on the target.
+          strip_command: Optional (default True). Strip command name from stdout.
+          delimiter: Optional. Delimiter to look for in the output to know when to stop expecting more output
+          (usually the shell prompt)
+          strip_delimiter: Optional (default True): Strip the provided delimiter from the output
+
+        Returns:
+          The stdout from the shell command.
+        """
+
+        conn = self._get_service_connection(b'shell:')
+
+        return self.protocol_handler.InteractiveShellCommand(conn, command=command, strip_command=strip_command,
```

Comment:
Can you bring this under 80 chars easily?

Reply:
Happy to make these formatting changes so its easier to see the diffs, give me a day or so to revisit it and I'll push up the changes.